src/module.c

```diff
@@ -1032,11 +1032,29 @@ int isCommandNameValid(const char *name) {
     return 1;
 }
 
+/* Helper for commandFlagsFromString(). Turns a string representing command
+ * flags into the ACL Category Flags.
+ *
+ * Returns 'REDISMODULE_OK' if acl category flag is recognized or
+ * returns 'REDISMODULE_ERR' if not recognized  */
+int matchAclCategoriesFlags(char *flag, int64_t *acl_categories){
+    if (flag[0] == '@'){
+        uint64_t this_flag = ACLGetCommandCategoryFlagByName(flag + 1);
+        if (this_flag) {
+            *acl_categories |= (int64_t) this_flag;
+            serverLog(LL_NOTICE,"Recognized acl category flag %s on module load", flag);
+            return REDISMODULE_OK;
+        }
+    }
+    return REDISMODULE_ERR; /* Unrecognized */
+}
+
 /* Helper for RM_CreateCommand(). Turns a string representing command
- * flags into the command flags used by the Redis core.
+ * flags into the command flags used by the Redis core and acl category flags used by Redis ACL
+ * which allows users to access the module commands by acl categories.
  *
  * It returns the set of flags, or -1 if unknown flags are found. */
-int64_t commandFlagsFromString(char *s) {
+int64_t commandFlagsFromString(char *s, int64_t *acl_categories) {
```

Comment:
maybe resetting `acl_categories` to 0 should be the responsibility of this function and not the caller (being an output parameter).